adminSiteServer/adminRouter.tsx

```diff
@@ -297,19 +309,21 @@ adminRouter.get(`/${EXPLORERS_PREVIEW_ROUTE}/:slug`, async (req, res) => {
         )
             return `File not found`
         const explorer = await explorerAdminServer.getExplorerFromFile(filename)
-        return renderExplorerPage(explorer, knex)
-    })
+        const explorerPage = renderExplorerPage(explorer, knex)
 
-    res.send(explorerPage)
-})
+        return res.send(explorerPage)
+    }
+)
 
-adminRouter.get("/datapage-preview/:id", async (req, res) => {
-    const variableId = expectInt(req.params.id)
-    const variableMetadata = await getVariableMetadata(variableId)
-    if (!variableMetadata) throw new JsonError("No such variable", 404)
+getPlainRouteWithROTransaction(
+    adminRouter,
+    "/datapage-preview/:id",
```

Comment:
I'm getting an error when trying to visit this route: http://localhost:3030/admin/datapage-preview/7216 which requires the server to be restarted

`error: Error: Transaction query already complete, run with DEBUG=knex:tx for more info`

Reply:
Thanks for catching that, fixed also in the top PR